build-tools/heroku/generate-rules-highlighting.py

```diff
@@ -0,0 +1,452 @@
+
+# extension of \w
+# A-z is different from A-Za-z (see an ascii table)
+CHARACTER = '0-9_A-Za-zÀ-ÿء-ي' 
+WORD      = '[' + CHARACTER + "]+"
+SPACE     = " +"
+
+START_LINE = '(^ *)'
+START_WORD = '(^| )'
+END_WORD   = '(?![' + CHARACTER + '])'
+
+
+# In the first levels, the strings are not yet well defined,
+# so we have to color them with respect to what is around,
+# so we use particular functions
+
+def rule_level1(keywordLang):
+    return [{
+        'regex': START_LINE + "(" + keywordLang["ASK"] + ")(.*)$",
+        'token': ['text','keyword','text'],
+        'next': 'start',
```

Comment:
What we had in the original TypeScript is a more maintainable abstraction over these raw rules.

What you have now is technically correct, but not as easy to change anymore as what we had. It's not easy to see what the differences between levels are anymore, for example.

Can you introduce a form of abstraction that improves the maintainability here?

(Put another way: yes there is JSON and regexes involved. I would prefer to see and edit as little of either as possible!)

Reply:
This file is also pretty large. It would be nice to split out the *mechanism* that allows us to write these rules and renders them to JSON, from the *actual rules* themselves.

The rules themselves probably also should not live in a directory called `build-tools`. They're part of Hedy proper, not just "some script that moves data from A to B".

So I would expect all of this code to live somewhere more prominent, and have the scripts here only invoke files from the other places as needed.

Reply:
Yes agreed! We talked about this yesterday and I would prefer all code related to highlighting to live in a folder together.

Reply:
It's done ! I created a folder with all the syntax highlighting stuff and split the python file (and add a readme)